src/components/DataTable/DataTable.vue

```diff
@@ -0,0 +1,319 @@
+    }
+  },
+  computed: {
+    internalSortByColumn(): DataTableColumn {
+      return this.columns.find(column => column.value == this.internalSortBy)!
+    },
+    searchableColumns(): DataTableColumn[] {
+      return this.columns.filter(column => column.search)
+    },
+    searched(): DataTableRow[] {
+      if(this.searchableColumns.length == 0) {
+        return this.rows
+      }
+
+      const searchable = this.searchableColumns.map(column => column.value)
+      const term = this.search.toLowerCase()
+
+      return this.rows.filter(row => {
+        const onlySearchableRow = pick(row, searchable)
+        const onlySearchableValues = Object.values(onlySearchableRow)
+
+        return onlySearchableValues.some(value => value.toString().toLowerCase().includes(term))
+      })
+    },
+    sorted(): DataTableRow[] {
+      const rows = this.searched
+
+      if(this.internalSortBy === null || this.internalDirection === 'none') {
+        return rows
+      }
+
+      const sorter = this.internalSortByColumn.sort ?? this.getDefaultSorter(this.internalSortBy)
+      const sorted = [...rows].sort(sorter)
+
+      if(this.internalDirection == 'desc') {
+        sorted.reverse()
+      }
+
+      return sorted
+    },
+  },
+  methods: {
+    setDirection(direction: DataTableColumnSort) {
+      this.internalDirection = direction
+      this.$emit('update:direction', direction)
+    },
+    setSortBy(column: DataTableColumn) {
+      this.internalSortBy = column.value
+      this.$emit('update:sortBy', column.value)
+    },
+    getDefaultSorter(sortBy: string) {
+      return (a: DataTableRow, b: DataTableRow) => a[sortBy].toString().localeCompare(b[sortBy].toString())
+    },
+    getColumnSortIconClasses(column: DataTableColumn)  {
+      if(column.value !== this.internalSortBy || this.internalDirection == 'none') {
+        return 'pi-code-line data-table__table-header-sort-icon--rotate'
+      }
+
+      return this.internalDirection == 'asc' ? 'pi-arrow-up-line' : 'pi-arrow-down-line'
+    },
+    sortColumn(column: DataTableColumn): void {
+      const sameColumn = this.internalSortBy == column.value
+      let direction: DataTableColumnSort = 'asc'
+
+      if(sameColumn) {
+        switch(this.internalDirection) {
+          case 'asc':
+            direction = 'desc'
+            break
+          case 'desc':
+            direction = 'none'
+            break
+          case 'none':
+            direction = 'asc'
+            break
+        }
+      }
+
+      this.setDirection(direction)
+      this.setSortBy(column)
+    },
+    clearSearch(): void {
+      this.search = ''
+    },
+    columnSlotName(column: DataTableColumn): string {
+      return `column-${kebabCase(column.value)}`
+    },
+    columnHeaderSlotName(column: DataTableColumn): string {
+      return `column-header-${kebabCase(column.value)}`
+    }
+  }
+})
+</script>
+
+<style scoped>
+
+.data-table {
+  border-spacing: 0;
+  width: 100%;
+  color: #465968;
```

Comment:
should these colors come from custom properties?

Reply:
Either custom properties or sass variables. Custom properties are prefered IMO. @ThatGalNatalie can you address this comment?

Reply:
Sure. I'll update those to use custom properties